[DOCS] Update put mapping (#120248)
elastic/elasticsearch: docs/reference/indices/put-mapping.asciidoc

For the most up-to-date API details, refer to {api-es}/group/endpoint-indices[Index APIs].
--

> You can use this API to add new fields to an existing index. However, Elasticsearch does not allow in-place field type changes. If you need to change a field’s type, you must create a new index with the updated mapping and reindex the data. 

Reviewer: ```suggestion
You can use this API to:

* Add new fields to an existing index
* Change the search settings of existing fields

NOTE: Elasticsearch does not allow in-place field type changes. If you need to change a field’s type, you must create a new index with the updated mapping and <<docs-reindex,reindex>> the data. 
```

> Changing field types is often done to improve query performance, enhance search relevance, or resolve data type mismatches (e.g., converting a text field to a keyword field for faster aggregations). However, if the existing data is malformed or incompatible with the updated field type, such as attempting to convert a text field to a date field, it may result in data loss during ingestion.

Reviewer: ```suggestion
```

I actually think this is the wrong spot for this guidance, we've just said you can't change a type using this API, so this belongs elsewhere.

> You can also use this API to change the search settings of existing fields.

Reviewer: ```suggestion
```
